<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Compare co-expression neighborhoods across species — compare_neighborhoods • rcomplex</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Compare co-expression neighborhoods across species — compare_neighborhoods"><meta property="og:description" content="For each ortholog pair, tests whether co-expression neighborhoods are
conserved between species using hypergeometric tests in both directions."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rcomplex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/paliocha/rcomplex/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Compare co-expression neighborhoods across species</h1>
    <small class="dont-index">Source: <a href="https://github.com/paliocha/rcomplex/blob/v0.1.0-beta1/R/comparison.R" class="external-link"><code>R/comparison.R</code></a></small>
    <div class="hidden name"><code>compare_neighborhoods.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>For each ortholog pair, tests whether co-expression neighborhoods are
conserved between species using hypergeometric tests in both directions.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">compare_neighborhoods</span><span class="op">(</span><span class="va">net1</span>, <span class="va">net2</span>, <span class="va">orthologs</span>, n_cores <span class="op">=</span> <span class="fl">1L</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-net-">net1<a class="anchor" aria-label="anchor" href="#arg-net-"></a></dt>
<dd><p>Network object for species 1 (output of <code><a href="compute_network.html">compute_network()</a></code>).</p></dd>


<dt id="arg-net-">net2<a class="anchor" aria-label="anchor" href="#arg-net-"></a></dt>
<dd><p>Network object for species 2 (output of <code><a href="compute_network.html">compute_network()</a></code>).</p></dd>


<dt id="arg-orthologs">orthologs<a class="anchor" aria-label="anchor" href="#arg-orthologs"></a></dt>
<dd><p>Data frame with columns <code>Species1</code>, <code>Species2</code>, and
<code>OrthoGroup</code> (output of <code><a href="parse_orthologs.html">parse_orthologs()</a></code>).</p></dd>


<dt id="arg-n-cores">n_cores<a class="anchor" aria-label="anchor" href="#arg-n-cores"></a></dt>
<dd><p>Number of threads for parallel computation (default 1).</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A data frame with columns:</p><dl><dt>Species1</dt>
<dd><p>Gene identifier for species 1</p></dd>

<dt>Species2</dt>
<dd><p>Gene identifier for species 2</p></dd>

<dt>OrthoGroup</dt>
<dd><p>Ortholog group identifier</p></dd>

<dt>Species1.neigh</dt>
<dd><p>Number of neighbors of Species1 gene in net1</p></dd>

<dt>Species1.ortho.neigh</dt>
<dd><p>Number of ortholog-mapped neighbors from net2</p></dd>

<dt>Species1.neigh.overlap</dt>
<dd><p>Intersection size</p></dd>

<dt>Species1.p.val.con</dt>
<dd><p>Upper-tail hypergeometric p-value for
conservation (direction 1)</p></dd>

<dt>Species1.p.val.div</dt>
<dd><p>Lower-tail hypergeometric p-value for
divergence (direction 1)</p></dd>

<dt>Species1.effect.size</dt>
<dd><p>Fold enrichment (direction 1). Values &gt; 1
indicate conservation, &lt; 1 indicate divergence.</p></dd>

<dt>Species2.neigh</dt>
<dd><p>Number of neighbors of Species2 gene in net2</p></dd>

<dt>Species2.ortho.neigh</dt>
<dd><p>Number of ortholog-mapped neighbors from net1</p></dd>

<dt>Species2.neigh.overlap</dt>
<dd><p>Intersection size</p></dd>

<dt>Species2.p.val.con</dt>
<dd><p>Upper-tail hypergeometric p-value for
conservation (direction 2)</p></dd>

<dt>Species2.p.val.div</dt>
<dd><p>Lower-tail hypergeometric p-value for
divergence (direction 2)</p></dd>

<dt>Species2.effect.size</dt>
<dd><p>Fold enrichment (direction 2). Values &gt; 1
indicate conservation, &lt; 1 indicate divergence.</p></dd>


</dl></div>
    <div id="ortholog-preparation-for-de-novo-transcriptomes">
    <h2>Ortholog preparation for de novo transcriptomes</h2>


<p>When working with de novo transcriptome assemblies, it is <strong>critical</strong> to
filter ortholog tables to expressed genes <strong>before</strong> applying any
maximum-paralogs filter. De novo assemblies can produce 10–50x more gene
models than are actually expressed (e.g. 205K models but only 14K expressed).
Unexpressed fragments inflate paralog counts per ortholog group, causing the
paralogs filter to discard groups that would pass if only expressed members
were counted. Failing to do this can reduce the usable gene set by 40–70%
and eliminate all statistical power from the hypergeometric test.</p>
<p>Recommended workflow:</p><ol><li><p>Build expression matrices for both species</p></li>
<li><p>Filter ortholog table to genes present in expression data</p></li>
<li><p>Apply maximum-paralogs filter on the reduced ortholog table</p></li>
<li><p>Compute networks and run comparison</p></li>
</ol></div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Martin Paliocha.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

      </footer></div>






  </body></html>

